app/src/main/java/com/github/se/eventradar/model/repository/event/FirebaseEventRepository.kt

```diff
@@ -118,4 +120,65 @@ class FirebaseEventRepository(db: FirebaseFirestore = Firebase.firestore) : IEve
 
     awaitClose { listener.remove() }
   }
+
+  override suspend fun addAttendee(eventId: String, attendeeUserId: String): Resource<Unit> {
+    return try {
+      eventRef.document(eventId).update("attendees_list", FieldValue.arrayUnion(attendeeUserId))
+      Resource.Success(Unit)
+    } catch (e: Exception) {
+      Resource.Failure(e)
+    }
+  }
+
+  override suspend fun removeAttendee(eventId: String, attendeeUserId: String): Resource<Unit> {
+    return try {
+      eventRef.document(eventId).update("attendees_list", FieldValue.arrayRemove(attendeeUserId))
+      Resource.Success(Unit)
+    } catch (e: Exception) {
+      Resource.Failure(e)
+    }
+  }
+
+  override suspend fun incrementPurchases(eventId: String): Resource<Unit> {
+    try {
+      db.runTransaction { transaction ->
+        val ref = eventRef.document(eventId)
+        val resultDocument = transaction.get(ref)
+        val purchases = resultDocument.getLong("ticket_purchases")
+        val capacity = resultDocument.getLong("ticket_capacity")
+
+        if (purchases != null && capacity != null) {
+          if (purchases < capacity) {
+            transaction.update(ref, "ticket_purchases", purchases + 1)
+          } else {
+            throw FirebaseFirestoreException(
+                "No more available tickets",
+                FirebaseFirestoreException.Code.ABORTED,
+            )
+          }
+        } else {
+          throw FirebaseFirestoreException(
+              "Invalid data",
+              FirebaseFirestoreException.Code.ABORTED,
+          )
+        }
+      }
+      return Resource.Success(Unit)
+    } catch (e: Exception) {
+      return Resource.Failure(e)
+    }
+  }
+
+  override suspend fun decrementPurchases(eventId: String): Resource<Unit> {
+    return addSubPurchases(eventId, -1)
```

Comment:
question: was there a reason you separated the function into `addSubPurchases` and `decrementPurchases` instead of including it as one?

Reply:
oh right, there was initially a reason. When I first coded it, incrementPurshases and decrementPurshases where using the same logic so this is why I separated. But afterwards I realized I needed a check on the ticket capacity. So now the addSubPurchases doesn't make sense anymore. I will remove it. Thanks for pointing this out !